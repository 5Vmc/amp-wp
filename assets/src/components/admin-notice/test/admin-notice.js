/**
 * External dependencies
 */
import { act } from 'react-dom/test-utils';
import { create } from 'react-test-renderer';

/**
 * WordPress dependencies
 */
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import {
	AdminNotice,
	ADMIN_NOTICE_TYPE_INFO,
	ADMIN_NOTICE_TYPE_SUCCESS,
	ADMIN_NOTICE_TYPE_WARNING,
	ADMIN_NOTICE_TYPE_ERROR,
} from '..';

let container;

describe( 'AdminNotice', () => {
	beforeEach( () => {
		container = document.createElement( 'div' );
		document.body.appendChild( container );
	} );

	afterEach( () => {
		document.body.removeChild( container );
		container = null;
	} );

	it( 'matches the snapshot', () => {
		const wrapper = create( <AdminNotice /> );

		expect( wrapper.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders a plain admin notice', () => {
		act( () => {
			render(
				<AdminNotice>
					{ 'Content' }
				</AdminNotice>,
				container,
			);
		} );

		expect( container.querySelector( '.admin-notice' ) ).not.toBeNull();
		expect( container.querySelector( '.admin-notice' ).textContent ).toBe( 'Content' );
	} );

	it( 'renders a dismissable admin notice', () => {
		const onDismiss = jest.fn();

		act( () => {
			render(
				<AdminNotice isDismissible={ true } onDismiss={ onDismiss } />,
				container,
			);
		} );

		expect( container.querySelector( '.admin-notice--dismissible' ) ).not.toBeNull();
		expect( container.querySelector( '.admin-notice__dismiss' ) ).not.toBeNull();

		act( () => {
			container.querySelector( '.admin-notice__dismiss' ).click();
		} );

		expect( onDismiss ).toHaveBeenCalledTimes( 1 );
	} );

	it.each( [
		[ ADMIN_NOTICE_TYPE_INFO ],
		[ ADMIN_NOTICE_TYPE_SUCCESS ],
		[ ADMIN_NOTICE_TYPE_WARNING ],
		[ ADMIN_NOTICE_TYPE_ERROR ],
	] )( 'renders a "%s" admin notice type', ( type ) => {
		act( () => {
			render(
				<AdminNotice type={ type } />,
				container,
			);
		} );

		expect( container.querySelector( `.admin-notice--${ type }` ) ).not.toBeNull();
	} );
} );
