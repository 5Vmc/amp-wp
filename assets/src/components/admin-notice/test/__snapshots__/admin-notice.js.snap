// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the AdminNotice component matches the snapshot 1`] = `
<div
  className="admin-notice admin-notice--info"
/>
`;
